orkaudio/filters/LiveStream/LiveStreamFilter.cpp

```diff
@@ -0,0 +1,265 @@
+
+template <class T>
+std::string toString(const T &value)
+{
+	std::ostringstream os;
+	os << value;
+	return os.str();
+}
+
+LiveStreamFilter::LiveStreamFilter()
+{
+	LOG4CXX_INFO(s_log, "LiveStream New Instance Created");
+}
+
+LiveStreamFilter::~LiveStreamFilter()
+{
+	LOG4CXX_INFO(s_log, "LiveStream Instance Destroying");
+}
+
+FilterRef LiveStreamFilter::Instanciate()
+{
+	FilterRef Filter(new LiveStreamFilter());
+	return Filter;
+}
+
+void LiveStreamFilter::AudioChunkIn(AudioChunkRef &inputAudioChunk)
+{
+	m_outputAudioChunk = inputAudioChunk;
+	// int16_t pcmdata[BUFFER_SAMPLES];
+	CStdString logMsg;
+
+	if (inputAudioChunk.get() == NULL)
+	{
+		return;
+	}
+
+	if (inputAudioChunk->GetNumSamples() == 0)
+	{
+		return;
+	}
+
+	AudioChunkDetails outputDetails = *inputAudioChunk->GetDetails();
+	char *inputBuffer = (char *)inputAudioChunk->m_pBuffer;
+	int size = outputDetails.m_numBytes * 2;
+
+	//logMsg.Format("LiveStreamFilter AudioChunkIn Size: %d, Encoding: %s , RTP payload type: %s",size ,toString(outputDetails.m_encoding) , RtpPayloadTypeEnumToString(outputDetails.m_rtpPayloadType));
+	//LOG4CXX_INFO(s_log, logMsg);
+
+	// @param sound_format Format of SoundData. The following values are defined:
+	// 0 = Linear PCM, platform endian
+	// 1 = ADPCM
+	// 2 = MP3
+	// 3 = Linear PCM, little endian
+	// 4 = Nellymoser 16 kHz mono
+	// 5 = Nellymoser 8 kHz mono
+	// 6 = Nellymoser
+	// 7 = G.711 A-law logarithmic PCM
+	// 8 = G.711 mu-law logarithmic PCM
+	// 9 = reserved
+	// 10 = AAC
+	// 11 = Speex
+	// 14 = MP3 8 kHz
+	// 15 = Device-specific sound
+	// Formats 7, 8, 14, and 15 are reserved.
+	// AAC is supported in Flash Player 9,0,115,0 and higher.
+	// Speex is supported in Flash Player 10 and higher.
+
+	char sound_format = 9;
+	if (outputDetails.m_rtpPayloadType == pt_PCMU)
+		sound_format = 8;
+	else if (outputDetails.m_rtpPayloadType == pt_PCMA)
+		sound_format = 7;
+
+	// @param sound_rate Sampling rate. The following values are defined:
+	// 0 = 5.5 kHz
+	// 1 = 11 kHz
+	// 2 = 22 kHz
+	// 3 = 44 kHz
+	char sound_rate = 3;
+
+	// @param sound_size Size of each audio sample. This parameter only pertains to
+	// uncompressed formats. Compressed formats always decode
+	// to 16 bits internally.
+	// 0 = 8-bit samples
+	// 1 = 16-bit samples
+	char sound_size = 1;
+
+	// @param sound_type Mono or stereo sound
+	// 0 = Mono sound
+	// 1 = Stereo sound
+	//char sound_type = outputDetails.m_channel == 0 ? 0 : 1;
+	char sound_type = 1;
+
+	timestamp += 160; //Timestamp increment = clock frequency/frame rate
+					  //160 byte payload of G.711 has a packetization interval of 20 ms
+					  //For 1 second, there will be 1000ms / 20ms = 50 frames
+					  //Audio RTP packet timestamp incremental value = 8kHz / 50 = 8000Hz / 50 = 160
+
+	if (isFirstPacket)
+	{
+		headChannel = outputDetails.m_channel;
+		isFirstPacket = false;
+	}
+
+	if (outputDetails.m_channel == headChannel && status)
+	{
+		bufferQueue.push(inputBuffer);
```

Comment:
you might want to check current used size of bufferQueue before pushing into it.

Reply:
@kingster Need your help here. How should we decide upon the maxSize for bufferQueue?

Reply:
fixed